src/compiler/crystal/semantic/main_visitor.cr

```diff
@@ -1618,6 +1641,10 @@ module Crystal
     end
 
     def visit(node : Return)
+      if inside_ensure?
+        node.raise "can't return from ensure (see https://github.com/crystal-lang/crystal/issues/4470)"
```

Comment:
Please don't include links to github issues in compiler error messages.

(same goes for all errors below this)